src/Widgets/Sidebar/NotebookList.vala

```diff
@@ -0,0 +1,48 @@
+/*
+* Copyright (c) 2011-2016 Felipe Escoto (https://github.com/Philip-Scott/Notes-up)
+*
+* This program is free software; you can redistribute it and/or
+* modify it under the terms of the GNU General Public
+* License as published by the Free Software Foundation; either
+* version 2 of the License, or (at your option) any later version.
+*
+* This program is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+* General Public License for more details.
+*
+* You should have received a copy of the GNU General Public
+* License along with this program; if not, write to the
+* Free Software Foundation, Inc., 59 Temple Place - Suite 330,
+* Boston, MA 02111-1307, USA.
+*
+* Authored by: Felipe Escoto <felescoto95@hotmail.com>
+*/
+
+public class ENotes.NotebookList : Granite.Widgets.SourceList.ExpandableItem {
+    private Gtk.Menu menu;
+    private Gtk.MenuItem new_notebook_item;
+
+    public NotebookList (string name = "") {
+        this.name = name;
+        
+        setup_menu ();
```

Comment:
Instead of doing it in a function for setup_menu, you can use the `construct` keyword instead. Here's an example on how it works: 

https://github.com/Philip-Scott/Spice-up/blob/master/src/Widgets/SlideWidget.vala#L46